easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -162,7 +162,7 @@ def get_toolchain_hierarchy(parent_toolchain):
         for dep in parsed_ec['dependencies']:
             ec = robot_find_easyconfig(dep['name'], det_full_ec_version(dep))
             ec = process_easyconfig(ec, validate=False)[0]
-            alldeps = [ec['ec']['toolchain']] + ec['ec']['dependencies']
+            alldeps = [ec['ec']['toolchain']] + parsed_ec['dependencies']
```

Comment:
@ocaisa: the problem is that the previous line was taking the dependencies of the dependencies...

Say we're parsing `GCC-4.9.3-2.25.eb`; the dependencies are `GCCcore` and `binutils`, which are listed in `parsed_ec['dependencies']`

We also want to take into account the toolchain for each of these dependencies, so `[ec['ec']['toolchain']]`

We're adding the same list of deps over and over again now, so I need to change this to:

```
deps_tcs = [d['name'] for d in parsed_ec['dependencies']]
for dep in parsed_ec['dependencies']:
    # also add toolchains of each dep
```

or something like that (I'll look into it later)